src/langchain_google_firestore/vectorstores.py

```diff
@@ -0,0 +1,419 @@
+                collection or document reference to store the data.
+            embeddings (Embeddings): The embeddings to use for the vector store.
+            client (Optional[Client], optional): The Firestore client to use. If
+                not provided, a new client will be created. Defaults to None.
+            content_field (str, optional): The field name to store the content
+                data. Defaults to "content".
+            metadata_fields (Optional[List[str]], optional): The list of metadata
+                fields to store. Defaults to None.
+            ignore_metadata_fields (Optional[List[str]], optional): The list of
+                metadata fields to ignore. Defaults to None.
+            text_embedding_field (Optional[str], optional): The field name to
+                store the text embeddings. Defaults to "embeddings".
+            distance_strategy (Optional[DistanceStrategy], optional): The distance
+                strategy to use for calculating distances between vectors.
+                Defaults to DistanceStrategy.COSINE.
+
+        Raises:
+            ImportError: If the `firestore` package is not found.
+
+        Example:
+            .. code-block:: python
+
+            from langchain_google_firestore.vectorstores import Firestore
+            from langchain.embeddings import GooglePalmEmbeddings
+
+            embeddings = GooglePalmEmbeddings()
+            firestore_vecstore = Firestore(source="my_collection", embeddings=embeddings)
+        """
+        try:
+            from google.cloud import firestore
+        except ModuleNotFoundError as exc:
+            raise ImportError(IMPORT_ERROR_MSG) from exc
+
+        # Check if the client is provided, otherwise create a new client with
+        # the default client info.
+        self.client = client or firestore.Client()
+
+        client_agent = self.client._client_info.user_agent
+        if not client_agent:
+            self.client._client_info.user_agent = USER_AGENT
+        elif USER_AGENT not in client_agent:
+            self.client._client_info.user_agent = " ".join([client_agent, USER_AGENT])
+
+        if isinstance(source, str):
+            self.source = self.client.collection(source)
+        elif isinstance(source, DocumentReference):
+            self.source = source.parent
+        else:
+            self.source = source
+
+        self.embedding = embedding
+        self.content_field = content_field
+        self.metadata_fields = metadata_fields
+        self.ignore_metadata_fields = ignore_metadata_fields
+        self.text_embedding_field = text_embedding_field
+        self.distance_strategy = distance_strategy
+
+    @property
+    def embeddings(self) -> Optional[Embeddings]:
+        return self.embedding
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        collection: Optional[str] = None,
```

Comment:
Are we not using this argument here?

I see is in the conditional only. But I don't see usage afterwards